app/models/api/translation_memory.php

```diff
@@ -49,35 +49,23 @@
 
     /*
         We are only interested in target strings with keys in common with our
-        source strings. Not sending noise to getTranslationMemoryResults() has
-        a major performance and memory impact.
+        source strings.
     */
-    $target_strings = array_intersect_key(
-        Utils::getRepoStrings($request->parameters[4], $repository),
-        $source_strings
-    );
-
-    /*
-        We are not interested in keeping duplicate strings that have
-        different keys because this API does not take into account the
-        frequency of matches but the similarity of the strings.
-    */
-    $target_strings = array_unique($target_strings);
-
-    /*
-        The + operator is slightly faster than array_merge and also easier
-        to read. The functional difference doesn't matter in this case
-        (http://stackoverflow.com/questions/7059721/array-merge-versus/27717809#27717809)
-     */
-    $source_strings_merged += $source_strings;
-    $target_strings_merged += $target_strings;
+    $target_strings = Utils::getRepoStrings($request->parameters[4], $repository);
 
+    foreach ($source_strings as $key => $value) {
+        if (isset($target_strings[$key]) && ! empty($target_strings[$key])) {
+            $output[] = [
+                $value,
+                $target_strings[$key],
+            ];
+        }
```

Comment:
It works fine, my only nit would be to use explicit indexes to make the code clearer in the class (`$set['source']` vs `$set[0]`). Not sure if there's a memory impact from that.
```PHP
            $output[] = [
                'source' => $value,
                'target' => $target_strings[$key],
            ];
```